delete-user-data/functions/src/helpers.ts

```diff
@@ -25,3 +28,26 @@ export const getDatabaseUrl = (
 
   return `https://${selectedDatabaseInstance}.${selectedDatabaseLocation}.firebasedatabase.app`;
 };
+
+export const hasValidUserPath = async (
+  ref: DocumentReference,
+  path: string,
+  uid: string
+): Promise<boolean> => {
+  /** Check path for valid user id */
+  if (path.includes(uid)) return Promise.resolve(true);
+
+  /** Check to find valid field */
+  const snapshot = await ref.get();
+
+  if (snapshot.exists) {
+    for (const field of config.searchFields.split(",")) {
+      if (snapshot.get(new FieldPath(field)) === uid) {
```

Comment:
Would a includes check here be better too?

something like
`
if (snapshot.get(new FieldPath(field)).includes(uid)) {
..
`